src/main/java/org/openstreetmap/atlas/checks/validation/intersections/RelationBoundary.java

```diff
@@ -0,0 +1,50 @@
+package org.openstreetmap.atlas.checks.validation.intersections;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.openstreetmap.atlas.geography.atlas.items.AtlasEntity;
+import org.openstreetmap.atlas.geography.atlas.items.Relation;
+
+/**
+ * Contains BoundaryParts and map of relations. It is a main object for analysing
+ * BoundaryIntersectionCheck.
+ *
+ * @author srachanski
+ */
+public class RelationBoundary
```

Comment:
This doesn't need to happen now, but this would be a great candidate for a ComplexEntity in Atlas.